extensions/azurePublishNew/src/node/provisionService.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { IBotProject } from '@botframework-composer/types';
+
+import { getAppRegistrationProvisionService } from './azureResources/appRegistration';
+import { getWebAppProvisionService } from './azureResources/webApp';
+import { getBotChannelProvisionService } from './azureResources/botChannel';
+import { getAzureFunctionsProvisionService } from './azureResources/azureFunctions';
+import { getCosmosDbProvisionService } from './azureResources/cosmosDb';
+import { getLuisAuthoringProvisionService } from './azureResources/luisAuthoring';
+import { getLuisPredictionProvisionService } from './azureResources/luisPrediction';
+import { getBlogStorageProvisionService } from './azureResources/blobStorage';
+import { getQnAProvisionService } from './azureResources/qna';
+import { getAppServiceProvisionService } from './azureResources/servicePlan';
+
+// bot project => candidate resources => select & configure resources => order & provision
+
+export type ProvisionWorkingSet = Record<string, object>;
+
+export type ResourceProvisionService = {
+  getDependencies: () => string[];
+  getRecommendationForProject: (project: IBotProject) => 'required' | 'optional' | 'invalid';
```

Comment:
Consider 'none' or 'notAllowed'  - invalid implies something is wrong with user input or state.